service/history/workflow/workflow_task_state_machine.go

```diff
@@ -739,3 +751,21 @@ func (m *workflowTaskStateMachine) afterAddWorkflowTaskCompletedEvent(
 	m.ms.executionInfo.LastWorkflowTaskStartId = event.GetWorkflowTaskCompletedEventAttributes().GetStartedEventId()
 	return m.ms.addBinaryCheckSumIfNotExists(event, maxResetPoints)
 }
+
+func (m *workflowTaskStateMachine) emitWorkflowTaskAttemptStats(
+	attempt int32,
+) {
+	namespaceName := m.ms.GetNamespaceEntry().Name().String()
+	m.ms.metricsClient.Scope(
+		metrics.WorkflowContextScope,
+		metrics.NamespaceTag(namespaceName),
+	).RecordDistribution(metrics.WorkflowTaskAttempt, int(attempt))
+	if attempt >= int32(m.ms.shard.GetConfig().WorkflowTaskCriticalAttempts()) {
+		m.ms.logger.Warn("Critical attempts processing workflow task",
```

Comment:
we need to be carefully that this could potentially pollute our logging, might need to use throttle logger here.

Reply:
Makes sense. Updated to use throttled logger.